### PR TITLE
A few (hopefully) helpful changes

### PR DESCRIPTION
**Pull request by one000mph**
Hello! I was happy to find your WIP drone plugin a week ago or so. I've modified it for my own use and in the spirit of open-source I'd like to share these changes if you'd like to consider merging them.

**Key Update in this PR:**
- Use implicit $NOW_TOKEN for the secret (I was having trouble getting drone to recognize the explicit token parameter, maybe a naming conflict??)
- Use `deploy_name` instead of `name` (I believe that $PLUGIN_NAME represents the name of the plugin step and this was causing the parameter to be overwritten at runtime)
- Add team scope with the `--team` flag
- Add helpful logging messages to the plugin script
- Sanity checking and a tiny bit of code cleanup

### CODE REVIEW

**Comment by one000mph** [comment]
The latest commits from de8865f to present add the ability to clean up old deploys when setting an alias.

**Comment by lucaperret** [comment]
Hello, thanks for your PR, there are nice improvements !
Could you merge with the current master branch commit?

File: .drone.yml
Comment (lucaperret) [inline .drone.yml]: Could you put the lucap/drone-now repo for docker image ?

File: DOCS.md
Comment (lucaperret) [inline DOCS.md]: Could you let the ability to set the Now token within a property `now_token` ?

File: README.md
Comment (lucaperret) [inline README.md:45]: Could you add this parameter into the DOCS please

File: script.sh
Comment (lucaperret) [inline script.sh]: Is this `NOW_TOKEN_OPTION` can be factorised with `NOW_AUTH`

File: script.sh
Comment (lucaperret) [inline script.sh]: This `NOW_TEAM_OPTION` can be factorised with `NOW_AUTH`

File: script.sh
Comment (lucaperret) [inline script.sh]: This parameter is not mandatory, Now deploy the working directory if directory isn't set.

File: script.sh
Comment (lucaperret) [inline script.sh:62]: Could you add this parameter to the DOCS.

File: .drone.yml
Comment (one000mph) [reply .drone.yml]: sure that is fixed in the latest commit. I will have to switch back after merging though since that is my own drone config file.

**Comment by one000mph** [comment]
I don't have write access to your repo. Everything OK to merge?

**Comment by lucaperret** [comment]
Could you review my comment before merging...

**Comment by one000mph** [comment]
Yep. I saw your comment about the the docker repo here:  https://github.com/lucaperret/drone-now/pull/1/files/8145ea22d378dbc3d7d4ed29c3e16b0df897102a#diff-3216dfff0ed3e301453e6799e8c367e2L5. 

That is fixed in my latest commit a5a5c1f. Is there anything else?

**Comment by lucaperret** [comment]
Thanks for that,
I've also started a review, could you answer and push changes ?

**Review by lucaperret — CHANGES_REQUESTED** [review]
Here's the review

File: script.sh
Comment (lucaperret) [inline script.sh]: $NOW_AUTH isn't declared before

File: DOCS.md
Comment (lucaperret) [inline DOCS.md]: There's missing a new line upper.

File: DOCS.md
Comment (lucaperret) [inline DOCS.md]: the `directory` param can be removed:
```diff
 -    directory: public
```

File: DOCS.md
Comment (lucaperret) [inline DOCS.md]: Cleaning up doesn't come from this documentation.

File: script.sh
Comment (one000mph) [reply script.sh:62]: also added!

File: README.md
Comment (one000mph) [reply README.md:45]: added!

File: DOCS.md
Comment (one000mph) [reply DOCS.md]: sure. I like that idea. updated in a697c4f

File: DOCS.md
Comment (lucaperret) [inline DOCS.md]: Should have diff format

File: script.sh
Comment (one000mph) [reply script.sh]: you are right. I refactored a bit to make this optional. my paradigm is using the plugin for static sites so thanks for that feedback.

File: script.sh
Comment (one000mph) [reply script.sh]: not 100% sure if I was understanding you here, but take a look at my latest commit and let me know what you think. tried to clean up and condense the code as much as possible

File: DOCS.md
Comment (lucaperret) [inline DOCS.md:134]: You should add it's usage as as a first example configuration

**Review by lucaperret — CHANGES_REQUESTED** [review]
Here's my last feedback

File: script.sh
Comment (lucaperret) [reply script.sh]: good now

File: DOCS.md
Comment (one000mph) [reply DOCS.md]: fixed!

File: script.sh
Comment (one000mph) [inline script.sh]: Here the deploy command is not surrounded by a conditional as it was before. I can't think of a case when $NOW_AUTH wouldn't already exist, but it might be safer to do a second check for  `if [ -n "$NOW_AUTH" ]` here and throw a descriptive log message in the `else` case? If you have more ideas on this feel free to request more changes

File: DOCS.md
Comment (lucaperret) [inline DOCS.md]: Now will take the name of the working directory as the deployment name

File: DOCS.md
Comment (lucaperret) [inline DOCS.md]: There's a missing new line to seperate the introduction and the first simple configuration title

File: DOCS.md
Comment (lucaperret) [inline DOCS.md]: this line is already removed from the diff

File: DOCS.md
Comment (lucaperret) [inline DOCS.md]: the diff is only add cleanup

File: script.sh
Comment (lucaperret) [inline script.sh]: Is this line useful ?

**Review by lucaperret — CHANGES_REQUESTED** [review]
other changes

File: DOCS.md
Comment (one000mph) [reply DOCS.md]: pushed an update with more accurate description

File: script.sh
Comment (one000mph) [reply script.sh]: nope, just removed

**Comment by lucaperret** [comment]
Hi @one000mph, can't wait for merging this PR 👍 
Can you take into account my feedbacks ?

**Comment by one000mph** [comment]
Hi @lucaperret yep I believe that I've taken into account all your comments. Can you take another look?

**Review by lucaperret — APPROVED** [review]

**Comment by lucaperret** [comment]
Example configuration for Aliases and Domains, Cleaning Up Old Deployments, Path Aliases has a `npm` property, it should be removed from the diff in the Aliases and Domains example.
And I will merge !

**Comment by lucaperret** [comment]
Thanks @one000mph for your hard work!

**Comment by one000mph** [comment]
@lucaperret glad I could help contribute to this! Thank you for your feedback. It will be important to push a build of this plugin to the docker repo `lucap/drone-now` so that others can use it in their pipeline.

**Comment by lucaperret** [comment]
Done!
https://github.com/lucaperret/drone-now/releases/tag/3.0.0
https://hub.docker.com/r/lucap/drone-now/tags/

**Pull request merged**
